Gems/Atom/Asset/Shader/Code/Source/Editor/ShaderVariantListBuilder.cpp

```diff
@@ -0,0 +1,514 @@
+                    }
+                }
+            }
+
+            return sourceShaderAbsolutePath;
+        }
+
+        //! Validates if a given .shadervariantlist file is located at the correct path for a given .shader full path.
+        //! There are two valid paths:
+        //! 1- Lower Precedence: The same folder where the .shader file is located.
+        //! 2- Higher Precedence: <project-path>/ShaderVariants/<Same Scan Folder Subpath as the .shader file>.
+        //! The "Higher Precedence" path gives the option to game projects to override what variants to generate. If this
+        //!     file exists then the "Lower Precedence" path is disregarded.
+        //! A .shader full path is located under an AP scan folder.
+        //! Example: "@gemroot:Atom_Feature_Common@/Assets/Materials/Types/StandardPBR_ForwardPass.shader"
+        //!     - In this example the Scan Folder is "<atom-gem-path>/Feature/Common/Assets", while the subfolder is "Materials/Types".
+        //! The "Higher Precedence" expected valid location for the .shadervariantlist would be:
+        //!     - <GameProject>/ShaderVariants/Materials/Types/StandardPBR_ForwardPass.shadervariantlist.
+        //! The "Lower Precedence" valid location would be:
+        //!     - @gemroot:Atom_Feature_Common@/Assets/Materials/Types/StandardPBR_ForwardPass.shadervariantlist.
+        //! @shouldExitEarlyFromProcessJob [out] Set to true if ProcessJob should do no work but return successfully.
+        //!     Set to false if ProcessJob should do work and create assets.
+        //!     When @shaderVariantListAbsolutePath is provided by a Gem/Feature instead of the Game Project
+        //!     We check if the game project already defined the shader variant list, and if it did it means
+        //!     ProcessJob should do no work, but return successfully nonetheless.
+        static bool ValidateShaderVariantListFolder(const AZStd::string& shaderVariantListAbsolutePath,
+            const AZStd::string& shaderAbsolutePath, bool& shouldExitEarlyFromProcessJob)
+        {
+            AZStd::string scanFolderFullPath;
+            AZStd::string shaderProductFileRelativePath;
+            bool success = false;
+            AzToolsFramework::AssetSystemRequestBus::BroadcastResult(success
+                , &AzToolsFramework::AssetSystem::AssetSystemRequest::GenerateRelativeSourcePath
+                , shaderAbsolutePath, shaderProductFileRelativePath, scanFolderFullPath);
+            if (!success)
+            {
+                AZ_Error(ShaderVariantListBuilderName, false, "Couldn't get the scan folder for shader [%s]", shaderAbsolutePath.c_str());
+                return false;
+            }
+            AZ_Trace(ShaderVariantListBuilderName, "For shader [%s], Scan folder full path [%s], relative file path [%s]", shaderAbsolutePath.c_str(), scanFolderFullPath.c_str(), shaderProductFileRelativePath.c_str());
```

Comment:
I think you still need \n at the end for AZ_Trace but I haven't used it yet to confirm